### DISCUSSION TITLE
Is address recognition a suitable use case?

### DISCUSSION THREAD

**Discussion by lmores** [moj-analytical-services/splink#848] [Q&A] [answered]
I have never used splink before and I need to understand whether or not it is well-suited for my problem.
I have a (growing) dataset of physical addresses. Each record in the dataset is made of the following fields: `person_name`, `street_name`, `street_number`, `postal_code`, `city_name`, `subdivision_code`, `country_code`.
All of them are strings and they may contain typos and any other type of errors.

My goals are:
  a) identify and group records that actually refer to the same place/address,
  b) when new records are added, verify if they belong to an already existing cluster.

This seems to be a typical use case for splink, but reading the documentation more carefully I found out that splink assumes the mutual independence of the features and an example of dependence is the pair (postal code, city name).

How can I solve this problem? Is it enough to only use one among `postal_code` and `city_name` in my "settings.comparisons" (and maybe use a blocking rule on the other one?).

Do you see any other obvious shortcoming? Do you have specific suggestions?

Thank you a lot!

**Comment by RobinL** [comment, marked as answer]
Hello.  I think this is likely to work quite well.  As you say, there is an issue of mutual independence being violated, but in practice record linkage models are often fairly robust to violations in this assumption.  Especially since in this case you have person name, which is 'mostly' independent from the address information.  

Yes - I agree with your suggestion that it would be better not to include both postcode and city name.  I would recommend dropping city name.  Another option - which would be relevant if postcode is sometimes null - may be to have a single comparison which as a first comparison level tries to compare postcode, but if that's null, compares city instead.  

Depending on its definition I might also suggest dropping subdivision code from the model.

If you can, please let us know how you get on!
